src/Verify.MSTest.SourceGenerator.Tests/Tests.cs

```diff
@@ -0,0 +1,137 @@
+// These tests don't use Verify.SourceGenerator to avoid creating a circular dependency between the repos.
+
+using VerifyMSTest.SourceGenerator;
+using VerifyMSTest.SourceGenerator.Tests;
+
+public class Tests(ITestOutputHelper output)
+{
+    readonly UsesVerifyTestDriver testDriver = new();
+
+    async Task VerifyGenerator(GeneratorDriverResults results)
+    {
+        output.WriteLine($"First run of generators took: {results.FirstRun.TimingInfo.ElapsedTime}");
+        output.WriteLine($"Cached re-run of generators took: {results.CachedRun.TimingInfo.ElapsedTime}");
+
+        await Verify(results.FirstRun.RunResult.SelectGeneratedSources());
+
+        // Ensure cachability
```

Comment:
for my own education. what make it cacheable? is it equality? or hashcodes? or both?

Reply:
The full incremental source generator pipeline has a good document here: https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md#caching, and this blog post is also helpful and the source of some of these test helpers: https://andrewlock.net/creating-a-source-generator-part-10-testing-your-incremental-generator-pipeline-outputs-are-cacheable/

Internally, the ValuesProvider pipeline is using both equality and hash codes. It uses the hash code first as a cheap "might be equal" check, and then if the hash codes match, uses `Equals` to perform the full equality check. https://thomaslevesque.com/2020/05/15/things-every-csharp-developer-should-know-1-hash-codes/ is a good quick reference.